youtube_dl/YoutubeDL.py

```diff
@@ -1836,7 +1850,56 @@ def ensure_dir_exists(path):
 
         self._write_thumbnails(info_dict, filename)
 
-        if not self.params.get('skip_download', False):
+        # Write internet shortcut files
+        url_link = webloc_link = desktop_link = False
+        if self.params.get('writelink', False):
+            if sys.platform == "darwin":  # macOS.
+                webloc_link = True
+            elif sys.platform.startswith("linux"):
+                desktop_link = True
+            else:  # if sys.platform in ['win32', 'cygwin']:
+                url_link = True
+        if self.params.get('writeurllink', False):
+            url_link = True
+        if self.params.get('writewebloclink', False):
+            webloc_link = True
+        if self.params.get('writedesktoplink', False):
+            desktop_link = True
+
+        if url_link or webloc_link or desktop_link:
+            ascii_url = iri_to_uri(info_dict['webpage_url'])
+
+        def _write_link_file(extension, template, newline, embed_filename):
+            linkfn = replace_extension(filename, extension, info_dict.get('ext'))
+            if self.params.get('nooverwrites', False) and os.path.exists(encodeFilename(linkfn)):
+                self.to_screen('[info] Internet shortcut is already present')
+            else:
+                try:
+                    self.to_screen('[info] Writing internet shortcut to: ' + linkfn)
+                    with io.open(encodeFilename(to_high_limit_path(linkfn)), 'w', encoding='utf-8', newline=newline) as linkfile:
+                        template_vars = {'url': ascii_url}
+                        if embed_filename:
+                            template_vars['filename'] = linkfn[:-len(extension) - 1]
+                        linkfile.write(template % template_vars)
+                except (OSError, IOError):
+                    self.report_error('Cannot write internet shortcut ' + linkfn)
+                    return False
+            return True
+
+        if url_link:
+            if not _write_link_file('url',     URL_LINK_TEMPLATE,     '\r\n', embed_filename=False): return
+        if webloc_link:
+            if not _write_link_file('webloc',  WEBLOC_LINK_TEMPLATE,  '\n',   embed_filename=False): return
+        if desktop_link:
+            if not _write_link_file('desktop', DESKTOP_LINK_TEMPLATE, '\n',   embed_filename=True ): return
+
+        if self.params.get('skip_download', False):
+            # Regarding the download archive, consider internet shortcut creation in conjunction with the `--skip-download` switch as everything the user wants. (See also help for the`--download-archive` switch.)
+            if url_link or webloc_link or desktop_link:
+                self.record_download_archive(info_dict)
```

Comment:
Download archive behavior must not change, it must only take place after success of the actual download and post processing.